aibolit/patterns/assign_null_finder/assign_null_finder.py

```diff
@@ -1,15 +1,28 @@
 # SPDX-FileCopyrightText: Copyright (c) 2019-2025 Aibolit
 # SPDX-License-Identifier: MIT
-
-import re
 from typing import List
 
-from aibolit.utils.encoding_detector import read_text_with_autodetected_encoding
+from aibolit.ast_framework.ast import AST
+from aibolit.ast_framework.ast_node import ASTNode
+from aibolit.ast_framework.ast_node_type import ASTNodeType
+from aibolit.types_decl import LineNumber
 
 
 class NullAssignment:
-    def value(self, filename: str) -> List:
-        source_code = read_text_with_autodetected_encoding(filename)
-        pattern = r'[^=!><]=(\s)*null(\s)*;'
-        return [lineIndex + 1 for lineIndex, line in
-                enumerate(source_code.split('\n')) if re.search(pattern, line)]
+    def value(self, ast: AST) -> List[LineNumber]:
+        lines = set()
+        for node in ast.get_proxy_nodes(ASTNodeType.ASSIGNMENT, ASTNodeType.VARIABLE_DECLARATOR):
+            if self._is_null_assignment(node):
+                lines.add(node.line)
+        return sorted(lines)
+
+    def _is_null_assignment(self, node: ASTNode) -> bool:
+        return (
+            node.node_type == ASTNodeType.ASSIGNMENT and
```

Comment:
@AntonProkopyev I failed to identify the original reason for two loops. I believe that your original implementation was more straightforward than this. I suggest that this is reverted back.